### PR TITLE
Add authentication button to yadocs form

### PR DESCRIPTION
**Pull request by korvin89**
<details>
<summary>Add this part of UI:</summary>

Screen Recording 2023-12-29 at 10 40 24 2023-12-29 10_42_00

</details>

### CODE REVIEW

File: src/ui/units/connections/components/custom-forms/Yadocs/components/AdditionalTitleContent.tsx [hunk 1]
```diff
@@ -3,19 +3,39 @@ import React from 'react';
...
 
 import iconSync from '../../../../../../assets/icons/sync.svg';
 
 const b = block('conn-form-yadocs');
 const ICON_SIZE = 18;
 
-export const AdditionalTitleContent = () => {
+type Props = {
+    authorized: boolean;
+    clickLoginButton: (oauthToken: string) => void;
```
Comment (kuzmadom) [inline src/ui/units/connections/components/custom-forms/Yadocs/components/AdditionalTitleContent.tsx:19]: maybe **on**ClickLoginButton?

File: src/ui/registry/units/common/types/components/OAuthTokenButton.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import type {ButtonProps} from '@gravity-ui/uikit';
+
+export type OAuthTokenButtonProps = ButtonProps & {
+    onTokenChange: (token: string) => void;
+    application: any;
```
Comment (kuzmadom) [inline src/ui/registry/units/common/types/components/OAuthTokenButton.ts:5]: Why any? Maybe limit it to `string`?

File: src/ui/units/connections/components/custom-forms/Yadocs/components/AdditionalTitleContent.tsx (hunk 1 above)
Comment (korvin89) [reply src/ui/units/connections/components/custom-forms/Yadocs/components/AdditionalTitleContent.tsx:19]: Such naming is already [here](https://github.com/datalens-tech/datalens-ui/blob/4c0dabd8c115be056b81dc52b39aceffd4b51aae/src/ui/units/connections/components/custom-forms/Yadocs/containers/DocsListContainer.tsx#L50). Plus it's a transit option, I'm not sure it's worth calling it that.

File: src/ui/registry/units/common/types/components/OAuthTokenButton.ts (hunk 2 above)
Comment (korvin89) [reply src/ui/registry/units/common/types/components/OAuthTokenButton.ts:5]: String does not satisfy real type of OAuthButton from closed part of our code. I will think about it, maybe there is a better approach

**Review by kuzmadom — APPROVED** [review]

File: src/ui/units/connections/components/custom-forms/Yadocs/components/AdditionalTitleContent.tsx (hunk 1 above)
Comment (kuzmadom) [reply src/ui/units/connections/components/custom-forms/Yadocs/components/AdditionalTitleContent.tsx:19]: It seems to me that it is logical for internal handlers and for handlers for events coming from outside (in props) to have different prefixes

I'm expecting an onEvent in props and something like a handleEvent inside the component.

Comment (korvin89) [reply src/ui/units/connections/components/custom-forms/Yadocs/components/AdditionalTitleContent.tsx:19]: What about props that drills into a deep?

**Pull request merged**